code/modules/mob/living/status_procs.dm

```diff
@@ -344,6 +344,8 @@
 	return SetSleeping(max(sleeping, amount), updating, no_alert)
 
 /mob/living/SetSleeping(amount, updating = 1, no_alert = FALSE)
+	if(frozen) // If the mob has been admin frozen, sleeping should not be changeable
+		return FALSE
```

Comment:
Do not return value [without a clearly-defined semantics for the return value].
Nothing checks AdjustSleeping for whether it succeeded. However *some* things (alcohol and friends) have stuff like `update_flags |= M.AdjustSleeping(-2, FALSE)`. Your particular change happens to be a no-op, but if you returned `TRUE` on success, that would have gotten reinterpreted as "please update health since it likely changed due to metabolism"